src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosExceptionProcessor.cs

```diff
@@ -0,0 +1,59 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Net;
+using EnsureThat;
+using Microsoft.Azure.Documents;
+using Microsoft.Health.Abstractions.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Context;
+using Microsoft.Health.Fhir.CosmosDb.Features.Metrics;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Features.Storage
+{
+    public class CosmosExceptionProcessor : ICosmosExceptionProcessor
```

Comment:
Maybe just change `CosmosMetricProcessor` to `CosmosResponseProcessor` and combine this logic in there? They all seem to be related to processing the response from Cosmos DB anyways.